app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -134,6 +107,62 @@ function() {
         return $boltHelper->transmit('orders', $orderRequest);
     }
 
+
+    /**
+     * Get Promise to create an order token
+     *
+     * @param string  $checkoutType  'multi-page' | 'one-page' | 'admin' | 'firecheckout'
+     *
+     * @return string javascript Promise string used for initializing BoltCheckout
+     */
+    protected function getBoltOrderTokenPromise($checkoutType) {
+
+        if ( $checkoutType === self::CHECKOUT_TYPE_FIRECHECKOUT ) {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl('boltpay/order/firecheckoutcreate');
+            $validPreconditionsCheck = 'isFireCheckoutFormValid';
+            $postData = 'checkout.getFormData ? checkout.getFormData() : Form.serialize(checkout.form, true)';
+        } else if ( $checkoutType === self::CHECKOUT_TYPE_ADMIN ) {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl("adminhtml/sales_order_create/create/checkoutType/$checkoutType", true);
+            $validPreconditionsCheck = ( $checkoutType === self::CHECKOUT_TYPE_ADMIN ) ? 'isAdminFormValid' : 'true';
+            $postData = "'true'";
+        } else {
+            $checkoutTokenUrl = Mage::helper('boltpay/url')->getMagentoUrl("boltpay/order/create/checkoutType/$checkoutType");
+            $validPreconditionsCheck = 'true';
+            $postData = "'true'";
+        }
+
+        return <<<PROMISE
+                    new Promise( 
+                        function (resolve, reject) {
+                            var checkoutAjaxId = setInterval(
+                                function() {
+                                     if ($validPreconditionsCheck) {
```

Comment:
@daisy1754 wrote:
hmm, this seems hacky.. can we instead listen for form state change and call BoltCheckout.configure whenever needed? I assume we'll need that logic anyway to deal with change in form data?

Reply:
@daisy1754 ok.

Reply:
On second thought, the change that you ask for is more involved and I believe it should be handled in a separate pull request.  Particularly, that PR would involve changing the long-standing how and when to call `BoltCheckout.configure`.  This PR is already involved enough.
  
Also, I am not in agreement that this approach is hacky and would need to understand the end goal a bit better with a discussion.  The other approach will involve more code in more areas, utilizing the same strategy and I estimate will appear at least the same level of "hackiness" and probably more.  So, that is further reason for that to be dealt in it's own devoted pull request and would make this PR too far involved in another unrelated topic.

Reply:
As far as I can tell, current code execute the function every 300ms and check the condition `isReadyToCreateBoltOrder`. Can we instead explicitly call `BoltCheckout.configure` when you are ready to create Bolt order?

Reply:
done.

Reply:
@daisy1754 done.

Reply:
SetInterval removed